### PR TITLE
enable pilot config validation

### PR DESCRIPTION
**Pull request by ayj**
Add config validation webhook to Pilot e2e tests. Now that
https://github.com/kubernetes/kubernetes/issues/49987 is fixed
validation webhooks can be enable on GKE >= 1.8. The following changes
are made as part of this PR:

* Rename external-admission-webhook-gke-1-7-x-workaround.sh to
  generate-webhook-cert.sh and refactor use cfssl and k8s API to
  generate and sign webhook certificate with k8s CA.

* Refactored pilot and e2e test defaults to use `istio-pilot` instead
  of `istio-pilot-external` service name. The k8s api-server will
  invoke the istio-pilot's webhook service directly to validate
  config.

* Added test code to generate self-signed CA and signed server
  cert/key. This enables validation webhook in automated e2e testing.

ref: https://github.com/istio/istio/issues/1478

### CODE REVIEW

**Comment by ayj** [comment]
cc @wattli for cert management as defined by https://docs.google.com/document/d/1f61uHoPvozoVhlut6TG66L2XQ_DvFQWUUeFjHJeypHQ/edit#bookmark=id.74mjrvtdgjl4

cc @jmuk for equivalent mixer webhook validation integration

**Comment by jmuk** [comment]
So do we want to remove 1.7 workaround? I think the new script is a little different about the load balancer hack, which doesn't work with 1.7.x.

**Comment by ayj** [comment]
Anyone using the workaround on GKE 1.7 is already on an alpha cluster. They would be likely to upgrade to 1.8.x anyway since alpha clusters expire after 30 days. The updated script should work for non-GKE clusters >= 1.7 and GKE >= 1.8.

**Comment by jmuk** [comment]
ah, fair enough, thanks.

File: install/kubernetes/generate-webhook-cert.sh [hunk 1]
```diff
@@ -0,0 +1,128 @@
+#!/bin/sh
```
Comment (cmluciano) [inline install/kubernetes/generate-webhook-cert.sh]: Can you make this file executable?

File: install/kubernetes/generate-webhook-cert.sh [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+set -e
+
+function usage() {
```
Comment (cmluciano) [inline install/kubernetes/generate-webhook-cert.sh]: This function specification does not work. You can remove the function keyword and just have usage.

File: install/kubernetes/generate-webhook-cert.sh [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+
+       --service          Service name of webhook.
+       --namespace        Namespace where webhook service resides.
+       --secret           Secret name for CA certificate and server certificate/key pair.
+       --secret-namespace Namespace where secret should be created. Default
+EOF
+    exit 1
+}
+
+while [[ $# -gt 0 ]]; do
```
Comment (cmluciano) [inline install/kubernetes/generate-webhook-cert.sh]: I'm unable to get this to work with the following example
```
./generate-webhook-cert.sh --service test --namespace default --secret webhook --secret-namespace default
```

File: install/kubernetes/generate-webhook-cert.sh (hunk 1 above)
Comment (ayj) [reply install/kubernetes/generate-webhook-cert.sh]: Thanks, fixed. I also changed to `#!/bin/bash`.

File: install/kubernetes/generate-webhook-cert.sh (hunk 3 above)
Comment (ayj) [reply install/kubernetes/generate-webhook-cert.sh]: I'm unable to see or reproduce the error. Any additional information on what's not working here?

Comment (cmluciano) [reply install/kubernetes/generate-webhook-cert.sh]: ok I think I must have had a typo. It works with the latest patchset now.

**Pull request merged**